### ISSUE TITLE
TST settings page goes blank after enough section are expanded

### ISSUE THREAD

**Issue by bughit** [piroor/treestyletab#2298]
## Short description
I ran into this after trying to export settings, which required turning on debug mode and expanding "all configs" (please make this less cumbersome). Some other sections were already expanded. After this the whole page went blank and it's no longer possible to get back to a usable state.

I've been able to repro in a new profile.

@piroor Please provide directions for modifying TST settings via devtools (to turn off debug mode and collapse all sections)

## Steps to reproduce

1. Start Firefox with clean profile.
2. Install TST.
3. go to tst setting
4. expand all sections
5. when you get to development, check "debug mode"
6. expand "all configs"
7. the entire settings area goes blank
    ![image](https://user-images.githubusercontent.com/895588/58905741-7675cb80-86d8-11e9-996c-60efb2d24f47.png)

## Environment

 * Platform (OS): win7 sp1 x64
 * Version of Firefox: 67.0.1 x64
 * Version (or revision) of Tree Style Tab: 3.0.15

**Comment by piroor** [comment]
I cannot reproduce this problem on my environment (Windows 10, Nightly 69.0a1 and Firefox 67).

Does any scrolling restore vanished options UI? Could you inspect how them become invisible, via about:debugging? If there is no problem around applied styles and bounding boxes, I think that the problem possibly appears due to a bug of the display driver or Firefox itself.

**Comment by bughit** [comment]
> Does any scrolling restore vanished options UI? 

Mostly no, it does sometimes reveal one textarea
![image](https://user-images.githubusercontent.com/895588/58927372-4a7c3980-871c-11e9-8667-69bf8202a2a2.png)

> Could you inspect how them become invisible, via about:debugging?

That did not allow me to inspect the settings area. It inspects "background/background.html". If you know how to do it please share, might come in handy in the future.

> I think that the problem possibly appears due to a bug of the display driver or Firefox itself.

Turning off hardware acceleration fixes it, so probably yes, however I have never encountered a similar issue in any other context so I am not going to turn it off.

Please allow access to import/export without turning on debug mode and expanding "all configs". That would mitigate this problem and just logically export/import does not need to depend on debug mode and "all configs" expansion.

>> Please provide directions for modifying TST settings

I've since figured out how turn off debug mode via the tst icon, which restored the settings page. However would still like to know how to inspect/edit tst or any extension settings. Aren't they in indexdb? Do you know how to get at them?

**Comment by piroor** [comment]
> That did not allow me to inspect the settings area. It inspects "background/background.html". If you know how to do it please share, might come in handy in the future.

You need to choose `options/options.html` from the list:
![image](https://user-images.githubusercontent.com/70062/58935012-13069f00-87a7-11e9-80f1-421091c824de.png)

Then you can inspect the options page with the "Inspector" tab of the debugger.

> Turning off hardware acceleration fixes it, so probably yes, however I have never encountered a similar issue in any other context so I am not going to turn it off.

Thank you for testing. Then I think this problem should be tracked on bugzilla.mozilla.org, because such an misrendering problem can happen on any regular webpage.

> Please allow access to import/export without turning on debug mode and expanding "all configs".

I'm still negative to reveal the feature for regular use, due to some reasons:

* TST's options are synchronized via Firefox Sync. I recommend users to use Firefox Sync to share addon options between multiple environments, instead of using addon-specific import/export features.
  * Please note that Firefox itself doesn't provide importing/exporting configs, and Firefox Sync is recommended for the purpose.
* The importing/exporting feature is not perfect. Actually it doesn't operate keyboard shortcuts customized by the user.
* The importing/exporting feature is implemented as a part of the library `Options.js` https://github.com/piroor/webextensions-lib-options and not a feature of TST itself. The "All Configs" feature (and importing/exporting) is common on my addons which depend on the library.

So I need a clear reason about why the import/export feature have to be revealed despite my policy. Instead of just a personal motivation, I need a logical reason.

**Comment by bughit** [comment]
> So I need a clear reason about why the import/export feature have to be revealed despite my policy

It facilitates the reproduction and reporting of bugs. The first thing I do if an issue is not trivially reproducible in a new profile, is import my main profile tst settings.

This bug (which is not likely to get fixed) make this more cumbersome than it already is.

Keeping import/export under Development seems sufficient discouragement of its mainstream use. It does not need to be hidden behind two more doors (debug, "all configs").

Could you comment about the settings?

>>However would still like to know how to inspect/edit tst or any extension settings. Aren't they in indexdb? Do you know how to get at them?

**Comment by piroor** [comment]
OK, I agree that those buttons are too hidden. I've introduced changes to put them under the "Development" section directly.

> However would still like to know how to inspect/edit tst or any extension settings. Aren't they in indexdb? Do you know how to get at them?

Sorry I forgot to comment about this point.

There is no "standard" method to save/load options in WebExtensions addons - someone uses WebExtensions storage API, someone uses cookies, someone uses indexedDB directly. But I think that WebExtensions storage API (local, sync, and managed) looks best in these choices, and I ordinary use it.

Addons don't know how Firefox stores those values. Actually old Firefox used JSON and the backend was migrated to indexedDB as you know. However, I don't know the reason but indexedDB storages for WebExtensions API backend looks invisible via Firefox's debugger. So I gave up to inspect raw data, and instead I always access to those data via WebExtensions API.
